Kernel/LPOComparator.hpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * This file is part of the source code of the software program
+ * Vampire. It is protected by applicable
+ * copyright laws.
+ *
+ * This source code is distributed under the licence found here
+ * https://vprover.github.io/license.html
+ * and in the source directory
+ */
+
+#ifndef __LPOComparator__
+#define __LPOComparator__
+
+#include "Forwards.hpp"
+
+#include "LPO.hpp"
+
+namespace Kernel {
+
+using namespace Lib;
+using namespace std;
+
+/**
+ * Runtime specialized LPO ordering check, based on the LPO check
+ * that has quadratic time complexity @see LPO::lpo.
+ */
+class LPOComparator
+: public OrderingComparator
+{
+public:
+  LPOComparator(const LPO& lpo) : _lpo(lpo), _instructions(), _res(Instruction::BranchTag::T_JUMP) {}
+
+  /** This function performs the runtime specialization and creates a comparator object. */
+  static LPOComparator* create(TermList tl1, TermList tl2, const LPO& lpo);
```

Comment:
Right, but do you mean we should plainly do the preprocessing in the ctor? Not that it does anything illegal which shouldn't be done in a ctor. (Although now I realize that the entire `Induction` inference code is done in the ctor, so this should be familiar :-))

Reply:
Right - if sensible + possible I think it would make sense.

I thought you would need move semantics for the `unique_ptr` construction later but `make_unique` or something might do the right thing, and maybe this is move-safe anyhow.